processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewayConfiguratorImpl.java

```diff
@@ -8,44 +8,45 @@
 import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
 import com.redhat.service.smartevents.infra.models.gateways.Action;
 import com.redhat.service.smartevents.infra.models.gateways.Source;
-import com.redhat.service.smartevents.processor.sources.SourceResolver;
+import com.redhat.service.smartevents.processor.resolvers.SourceConnectorResolver;
+import com.redhat.service.smartevents.processor.validators.DefaultGatewayValidator;
+import com.redhat.service.smartevents.processor.validators.GatewayValidator;
+import com.redhat.service.smartevents.processor.validators.custom.CustomGatewayValidator;
 
 @ApplicationScoped
 public class GatewayConfiguratorImpl implements GatewayConfigurator {
 
     @Inject
-    Instance<GatewayValidator<Action>> actionValidators;
+    Instance<CustomGatewayValidator> customValidators;
+
     @Inject
     Instance<GatewayResolver<Action>> actionResolvers;
+
     @Inject
-    Instance<GatewayValidator<Source>> sourceValidators;
+    DefaultGatewayValidator defaultGatewayValidator;
 
     @Inject
-    SourceResolver sourceResolver;
+    SourceConnectorResolver sourceConnectorResolver;
 
     @Override
-    public GatewayValidator<Action> getActionValidator(String actionType) {
-        return getOptionalBean(actionValidators, actionType)
-                .orElseThrow(() -> new GatewayProviderException(String.format("No validator found for action type '%s'", actionType)));
+    public GatewayValidator getValidator(String actionType) {
+        Optional<CustomGatewayValidator> customValidator = getOptionalBean(customValidators, actionType);
+        if (customValidator.isPresent()) {
+            return customValidator.get();
+        }
+        return defaultGatewayValidator;
     }
```

Comment:
Basically the trick is here.. if a `CustomGatewayValidator` is found we return it. Otherwise we use the default one